app/coffee/BucketController.coffee

```diff
@@ -0,0 +1,29 @@
+settings = require("settings-sharelatex")
+logger = require("logger-sharelatex")
+FileHandler = require("./FileHandler")
+metrics = require("metrics-sharelatex")
+Errors = require('./Errors')
+
+module.exports = BucketController =
+
+	getFile: (req, res)->
+		{bucket} = req.params
+		key = req.params[0]
+		credentials = settings.filestore.s3?[bucket]
```

Comment:
Can we check that these credentials exist before passing them through to the rest of the stack.

Reply:
My intention was that if they didn't exist, it would fall back to the default s3 credentials. That logic is in `S3PersistorManager`:
```
	getFileStream: (bucketName, key, opts, callback = (err, res)->)->
		opts = opts || {}
		headers = {}
		if opts.start? and opts.end?
			headers['Range'] = "bytes=#{opts.start}-#{opts.end}"
		callback = _.once callback
		logger.log bucketName:bucketName, key:key, "getting file from s3"
		s3Client = knox.createClient
			key: opts.credentials?.auth_key || settings.filestore.s3.key
			secret: opts.credentials?.auth_secret || settings.filestore.s3.secret
			bucket: bucketName
```

Reply:
Ok, I didn't like that before but actually it is fair enough as it gives us the ability to use this endpoint for other things if we needed.